ocf_data_sampler/torch_datasets/sample/uk_regional.py

```diff
@@ -67,78 +85,170 @@ def validate_sample(self, config: Configuration) -> bool:
+                        f"Expected NWP providers are missing from the data: "
+                        f"{list(missing_expected_providers)}",
+                    )
 
+                for provider in expected_shapes[nwp_key]:
+                    provider_data = nwp_data_all_providers[provider]
+
+                    if "nwp" not in provider_data:
+                        error_msg = (
+                            f"Missing array key 'nwp' in NWP data for provider '{provider}'."
+                        )
+                        raise ValueError(error_msg)
+
+                    nwp_array = provider_data["nwp"]
                     check_dimensions(
-                        actual_shape=actual_shape,
-                        expected_shape=expected_shape,
+                        actual_shape=nwp_array.shape,
+                        expected_shape=expected_shapes[nwp_key][provider],
                         name=f"NWP data ({provider})",
                     )
+            else:
+                warning_info = validation_warning(
+                    message=(
+                        f"NWP data ('{nwp_key}') is present but not expected "
+                        "in configuration."
+                    ),
+                    warning_type="unexpected_component",
+                    component=str(nwp_key),
+                )
+                logger.warning(
+                    f"{warning_info['message']} (Type: {warning_info['type']}, "
+                    f"Component: {warning_info.get('component')})",
+                )
 
         # Validate satellite data
         sat_key = SatelliteSampleKey.satellite_actual
-        # Check if Satellite data is expected but missing
         if sat_key in expected_shapes and sat_key not in self._data:
             raise ValueError(f"Configuration expects Satellite data ('{sat_key}') but is missing.")
 
-        # Check satellite shape if data exists and is expected
-        if sat_key in self._data and sat_key in expected_shapes:
-            sat_data = self._data[sat_key]
-            check_dimensions(
-                actual_shape=sat_data.shape,
-                expected_shape=expected_shapes[sat_key],
-                name="Satellite data",
-            )
+        if sat_key in self._data:
+            if sat_key in expected_shapes:
+                sat_data = self._data[sat_key]
+                check_dimensions(
+                    actual_shape=sat_data.shape,
+                    expected_shape=expected_shapes[sat_key],
+                    name="Satellite data",
+                )
+            else:
+                warning_info = validation_warning(
+                    message=(
+                        f"Satellite data ('{sat_key}') is present but not expected "
+                        "in configuration."
+                    ),
+                    warning_type="unexpected_component",
+                    component=str(sat_key),
+                )
+                logger.warning(
```

Comment:
why not raise the logger.warning in `validation_warning`, then you dont have to repeat code?